src/basic/unit-file.c

```diff
@@ -273,6 +274,37 @@ int unit_file_build_name_map(
                         return log_oom();
         }
 
+        /* Go over all our search paths, chase their symlinks and store the
+         * result in the expanded_search_paths list.
+         *
+         * This is important for cases where any of the unit directories itself
+         * are symlinks into other directories and would therefore cause all of
+         * the unit files to be recognized as linked units.
+         *
+         * This is important for distributions such as NixOS where most paths
+         * in /etc/ are symlinks to some other location on the filesystem (e.g.
+         * into /nix/store/).
+         */
+        STRV_FOREACH(dir, (char**) lp->search_path) {
+                _cleanup_free_ char *resolved_dir = NULL;
+                r = strv_extend(&expanded_search_paths, *dir);
+                if (r < 0) {
+                        log_debug("Failed to add existing path to expanded search paths: %s", *dir);
```

Comment:
I could probably drop this debug message here. Not sure why I left this in. Not really adding much information in this error case IMO.